server/src/weaverbird/pipeline/pipeline.py

```diff
@@ -184,6 +187,79 @@ def dict(self, *, exclude_none: bool = True, **kwargs) -> dict:
 ]
 
 
+def _contains_void_as_value(value: dict) -> bool:
+    return any(
+        [
+            value.get("value") == "__VOID__",
+            value.get("column") == "__VOID__",
+        ]
+    )
+
+
+def _remove_void_value_elements(obj: dict[str, Any] | list[dict[str, Any]]) -> None:
```

Comment:
```suggestion
def _remove_void_value_elements(step: FilterStep) -> FilterStep | None:
```

where `None` means that no conditions are left in the step

Reply:
Thanks for the suggestion, but i don't get it well... i added a function called `_clean_filter_step` to handle the step cleaning here : https://github.com/ToucanToco/weaverbird/blob/79280735dd457e12924b9186f85236c29825232d/server/src/weaverbird/pipeline/pipeline.py#L233

Reply:
Yes but the first thing it does is converting the instance to as dict so it's still the same as before

Reply:
In this implementation, we need a recursive function to walk through the whole dict representation of a FIlterStep, searching for dict/list or embed list in dict inside the filter step whatever the depth, how can we do that without having the dict representation of the filter step... if you've tips for this signature change it could be great please 🙏🏽 , because am not seeing well how to update that.

```python
def _remove_void_value_elements(step: FilterStep) -> FilterStep | None:
```

Reply:
Is this approach better ? 
https://github.com/ToucanToco/weaverbird/pull/1763/commits/32babad231f03936a026e0ab3b4f74c8a0c6a7f8

Reply:
It's still working on a dict rather than on a `Condition`... But I understood this would be implemented in laputa anyway right ? no need to invest superfluous energy here

Reply:
https://github.com/ToucanToco/weaverbird/pull/1763/commits/e8970edc19feacfbeca8fd2db3bb46d02f643542